### DISCUSSION TITLE
issue: input doesn't pick up value after remount

### DISCUSSION THREAD

**Discussion by lukasgebhard** [react-hook-form/react-hook-form#10747] [Q&A]
### Version Number

7.45.2

### Codesandbox/Expo snack

https://codesandbox.io/s/frosty-bose-wn2fqh

### Steps to reproduce

1. Empty formField1.
2. Enter "1" into formField1.

### Expected behaviour

formField2 displays "2".

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by lukasgebhard** [comment]
Note that the data the form submits (after following the above steps) depends on the used version of react-hook-form:

- v7.38.0: `{formField1: "1", formField2: undefined}`
- v7.45.2: `{formField1: "1", formField2: "2"}`

In both cases, formField2 doesn't display any value, though.

**Comment by lukasgebhard** [comment]
Related issues:

- https://github.com/react-hook-form/react-hook-form/issues/10096
- https://github.com/react-hook-form/react-hook-form/issues/10700
- https://github.com/react-hook-form/react-hook-form/issues/10386
- https://github.com/react-hook-form/react-hook-form/issues/8669

**Comment by bluebill1049** [comment]
`shouldUnregister` with strict mode is troublesome, i recommend invoking unreigster manually at your own app level.

**Comment by lukasgebhard** [comment]
Hi @bluebill1049,

Thank you for the fast reply and all your work to maintain this project!

> shouldUnregister with strict mode is troublesome, i recommend invoking unreigster manually at your own app level.

As you can see in the provided code sandbox, the reported issue occurs without strict mode.

**Comment by bluebill1049** [comment]
it is `shouldUnregister={true}`, your component gets unmounted and after unmount. The data is destroyed. If you want to fine-tune the experience, I would recommend to manually invoke `unregister` api.

**Reply by lukasgebhard** [reply]
>  I would recommend to manually invoke unregister api.

In the given code sandbox example, how would that help to restore `formField2`'s default value?

**Reply by lukasgebhard** [reply]
Now I've found a way:

1. Remove `shouldUnregister={true}`
2. Call `unregister('formField2', {"keepDefaultValue": true})`

Here the full workaround: https://codesandbox.io/s/rhf-issue-workaround-qqmrj8

**Comment by lukasgebhard** [comment]
> after unmount. The data is destroyed.

But why does the form submit the default value then? See [my previous comment](https://github.com/orgs/react-hook-form/discussions/10747#discussioncomment-6618162)

In my view, it's clearly a bug that after re-mounting `formField2`, its default value is present but not rendered. Either the default value should be gone entirely, or it should be present and rendered properly.

**Reply by sKopheK** [reply]
i've recently bumped into few cases when unregistering field removes its default value (it's actually in the library code itself, not any side effect). when it's displayed and registered again, there's no value whatsover. why would the library remove anything from `defaultValues`? i'd like to prevent any fidling with such "constant" object using `Object.freeze`, but it'd cause error
